tests/test_feature_manager.cc

```diff
@@ -123,6 +117,126 @@ namespace rascal {
+    auto & inner_sizes = Fix::inner_sizes;
+    auto & representations = Fix::representations;
+
+    // build the feature managers. only the 1st
+    // one will be used
+    for (size_t i_hyper{0}; i_hyper < hypers.size(); i_hyper++) {
+      features.emplace_back(inner_sizes[i_hyper], hypers[i_hyper]);
+    }
+
+    using data_t = typename Fix::Representation_t::SparseProperty_t::data_t;
+    std::vector<data_t> original_data{};
+    // extract the feature matrices in a ref vector
+    for (size_t i_hyper{0}; i_hyper < hypers.size(); i_hyper++) {
+      original_data.emplace_back();
+      for (auto & representation : representations[i_hyper]) {
+        auto & raw_data{representation.get_representation_sparse_raw_data()};
+        original_data.back().insert(original_data.back().end(),
+                                    raw_data.begin(), raw_data.end());
+      }
+    }
+
+    // move the features into the feature manager
+    for (size_t i_hyper{0}; i_hyper < hypers.size(); i_hyper++) {
+      for (size_t i_rep{0}; i_rep < representations[i_hyper].size(); i_rep++) {
+        features[i_hyper].push_back(representations[i_hyper][i_rep]);
+      }
+    }
+
+    // check if the features have been moved properly
+    for (size_t i_hyper{0}; i_hyper < hypers.size(); i_hyper++) {
+      auto feature_matrix = features[i_hyper].get_feature_matrix_dense();
+      auto & inner_size{inner_sizes[i_hyper]};
+      std::set<std::vector<int>> unique_keys{};
+      for (size_t i_center{0}; i_center < original_data[i_hyper].size();
+           ++i_center) {
+        for (auto & element : original_data[i_hyper][i_center]) {
+          unique_keys.emplace(element.first);
+        }
+      }
+      for (size_t i_center{0}; i_center < original_data[i_hyper].size();
+           ++i_center) {
+        auto & datas{original_data[i_hyper][i_center]};
+        int i_count{0};
+        double diff{0};
+        if (verbose)
+          std::cout << "Center: " << i_center << std::endl;
+        for (auto & key : unique_keys) {
+          // if the key exist
+          if (datas.count(key) == 1) {
+            if (verbose)
+              std::cout << "Key: " << key[0] << std::endl;
+            auto & data = datas[key];
+            for (int i_col{0}; i_col < data.cols(); i_col++) {
+              for (int i_row{0}; i_row < data.rows(); i_row++) {
+                diff = data(i_row, i_col) - feature_matrix(i_count, i_center);
+                if (verbose) {
+                  if (diff > 1e-12) {
+                    std::cout << diff << ", ";
+                    std::cout << data(i_row, i_col) << ", ";
+                    std::cout << feature_matrix(i_count, i_center) << "/ ";
+                  }
+                }
+
+                i_count++;
+              }
+              if (verbose)
+                std::cout << std::endl;
+            }
+          } else {
+            i_count += inner_size;
+          }
+        }
+        BOOST_CHECK_LE(diff, 1e-11);
```

Comment:
Is it possible to add to this test now only columns/rows but access with clusters?
This would show that the map works?
Or am I missing something?